umh-core/pkg/communicator/actions/delete-dataflowcomponent.go

```diff
@@ -136,3 +148,27 @@ func (a *DeleteDataflowComponentAction) getUuid() uuid.UUID {
 func (a *DeleteDataflowComponentAction) GetComponentUUID() uuid.UUID {
 	return a.componentUUID
 }
+
+func (a *DeleteDataflowComponentAction) waitForComponentToBeRemoved() error {
+	//check the system snapshot and waits for the instance to be removed
+	ticker := time.NewTicker(1 * time.Second)
+	defer ticker.Stop()
+	timeout := time.After(constants.DataflowComponentWaitForActiveTimeout)
+	for {
+		select {
+		case <-timeout:
+			return fmt.Errorf("dataflowcomponent %s was not removed in time", a.componentUUID)
+		case <-ticker.C:
+			if dataflowcomponentManager, exists := a.systemSnapshot.Managers[constants.DataflowcomponentManagerName]; exists {
+				instances := dataflowcomponentManager.GetInstances()
+				for _, instance := range instances {
+					if dataflowcomponentconfig.GenerateUUIDFromName(instance.ID) == a.componentUUID {
+						// component is still there, so we need to wait longer
+						continue
+					}
+					return nil
```

Comment:
You probably want a found bool here and then return nil, when non of the instances matched